src/Libraries/RevitNodes/Elements/Parameter.cs

```diff
@@ -25,12 +34,168 @@ public string Name
             }
         }
 
+        /// <summary>
+        /// Get the value of the parameter
+        /// </summary>
+        public object Value
+        {
+            get { return Revit.Elements.InternalUtilities.ElementUtils.GetParameterValue(this.InternalParameter); }
+        }
+
+        /// <summary>
+        /// Check if the Parameter has a value
+        /// </summary>
+        public bool HasValue
+        {
+            get { return InternalParameter.HasValue; }
+        }
+
+        /// <summary>
+        /// Check if the Parameter is read only
+        /// </summary>
+        public bool IsReadOnly
+        {
+            get { return InternalParameter.IsReadOnly; }
+        }
+
+        /// <summary>
+        /// Check if the Parameter is shared
+        /// </summary>
+        public bool IsShared
+        {
+            get { return InternalParameter.IsShared; }
+        }
+
+        /// <summary>
+        /// Get the parameter's group
+        /// </summary>
+        public string Group
+        {
+            get { return InternalParameter.Definition.ParameterGroup.ToString(); }
+        }
+
+        /// <summary>
+        /// Get the parameter type
+        /// </summary>
+        public string ParameterType
+        {
+            get { return InternalParameter.Definition.ParameterType.ToString(); }
+        }
+
+        /// <summary>
+        /// Get the parameter's element Id
+        /// </summary>
+        public int Id
+        {
+            get { return InternalParameter.Id.IntegerValue; }
+        }
+
+        /// <summary>
+        /// Get the parameter's unit type
+        /// </summary>
+        public string UnitType
+        {
+            get { return InternalParameter.Definition.UnitType.ToString(); }
+        }
+
+
+        /// <summary>
+        /// Get Element's Parameter by Name
+        /// </summary>
+        /// <param name="element">Element</param>
+        /// <param name="name">Parameter Name</param>
+        /// <returns>Parameter</returns>
+        public static Elements.Parameter ParameterByName(Elements.Element element, string name)
+        {
+            var param = element.Parameters.Cast<Autodesk.Revit.DB.Parameter>()
+                .OrderBy(x => x.Id.IntegerValue)
+                .FirstOrDefault(x => x.Definition.Name == name);
+
+            if (param == null)
+            {
+                return null;
+            }
+            else
+            {
+                return new Parameter(param);
+            }
+        }
+
+        /// <summary>
+        /// Set the value of the parameter
+        /// </summary>
+        public static void SetValue(Parameter parameter, object value)
```

Comment:
In a way yes, but I'd say my implementation is more flexible and handles more cases (eg. if the parameter expects an ElementId you can submit an int or an Element. Same for Int, you can submit a double, int or string to get to the correct result.